consensus/src/consensus/mod.rs

```diff
@@ -442,29 +451,42 @@ impl<N: Network> Consensus<N> {
         None
     }
 
-    /// Requests heads from connected peers in a predefined interval.
+    /// Requests heads from connected peers.
     fn request_heads(&mut self) {
-        // If there's no ongoing head request and we have at least one peer, check whether we should
-        // start a new one.
-        if self.head_requests.is_none() && (self.num_agents() > 0 || self.min_peers == 0) {
-            // This is the case if `head_requests_time` is unset or the timeout is hit.
-            let should_start_request = self
-                .head_requests_time
-                .map(|time| time.elapsed() >= Self::HEAD_REQUESTS_TIMEOUT)
-                .unwrap_or(true);
-            if should_start_request {
-                debug!(
-                    "Initiating head requests (to {} peers)",
-                    self.sync.num_peers()
-                );
-                self.head_requests = Some(HeadRequests::new(
-                    self.sync.peers(),
-                    Arc::clone(&self.network),
-                    self.blockchain.clone(),
-                ));
-                self.head_requests_time = Some(Instant::now());
-            }
+        // Wait for an ongoing head request to finish.
+        if self.head_requests.is_some() {
+            return;
         }
+
+        // We need at least one synced peer to perform a head request.
+        // Specifying `min_peers = 0` in the consensus config allows the first seed node
+        // on a network to establish consensus without any other nodes present.
+        if self.num_agents() == 0 && self.min_peers > 0 {
+            return;
+        }
+
+        // This is the case if `head_requests_time` is unset or the timeout is hit.
+        let should_start_request = self
+            .head_requests_time
+            .map(|time| time.elapsed() >= Self::HEAD_REQUESTS_TIMEOUT)
+            .unwrap_or(true);
+        if !should_start_request {
+            return;
+        }
+
+        debug!(
+            "Initiating head requests (to {} peers)",
+            self.sync.num_peers()
+        );
+
+        self.head_requests = Some(HeadRequests::new(
+            self.sync.peers(),
+            Arc::clone(&self.network),
+            self.blockchain.clone(),
+        ));
+
+        self.head_requests_time = Some(Instant::now());
+        self.head_requests_interval = interval(Self::HEAD_REQUESTS_TIMEOUT);
```

Comment:
The first tick always completes immediately. 
I am not sure if that is a problem, I believe most likely not as it only triggers a wake, but if it is, we should skip the first instance.